packages/plugin/src/PivotPlugin.js

```diff
@@ -109,69 +140,52 @@ class PivotPlugin extends Component {
-                    this.props.d2,
-                    visualization,
-                    options
-                ));
-
-                extraOptions[BASE_FIELD_YEARLY_SERIES] = yearlySeriesLabels;
-                extraOptions.xAxisLabels = computeGenericPeriodNames(responses);
-            } else {
-                responses = await apiFetchAnalytics(
-                    this.props.d2,
-                    visualization,
-                    options
-                );
-            }
+            const responses = await apiFetchAnalytics(
+                this.props.d2,
+                visualization,
+                options
+            );
 
             if (responses.length) {
                 onResponsesReceived(responses);
             }
 
             this.recreateVisualization = animation => {
-                const visualizationConfig = createVisualization(
-                    responses,
+                const remappedOptions = {
+                    showColTotals: visualization.colTotals,
+                    showRowTotals: visualization.rowTotals,
+                    showColSubTotals: visualization.colSubTotals,
+                    showRowSubTotals: visualization.rowSubTotals,
+                    numberType: visualization.numberType || 'VALUE', // TODO read default from options, perhaps better idea is to compute layout content in app
+                };
+
+                const layout = new d2aLayout(
+                    refs,
                     visualization,
-                    this.canvasRef.current,
-                    {
-                        ...extraOptions,
-                        animation,
-                    },
-                    undefined,
-                    undefined,
-                    isSingleValue(visualization.type) ? 'dhis' : 'highcharts' // output format
+                    remappedOptions
                 );
 
-                if (isSingleValue(visualization.type)) {
-                    onChartGenerated(visualizationConfig.visualization);
-                } else {
-                    onChartGenerated(
-                        visualizationConfig.visualization.getSVGForExport({
-                            sourceHeight: 768,
-                            sourceWidth: 1024,
-                        })
-                    );
-                }
+                const extraOptions = { renderLimit: 100000, trueTotals: true };
+
+                const pivotTable = new d2aTable.PivotTable(
+                    refs,
+                    layout,
+                    new d2aResponse(refs, responses[0].response),
+                    extraOptions
+                );
+
+                pivotTable.initialize();
+
+                pivotTable.build();
+
+                this.canvasRef.current.innerHTML = pivotTable.render();
             };
 
             this.recreateVisualization();
 
             this.setState({ isLoading: false });
         } catch (error) {
+            console.log(error);
```

Comment:
For testing only?

Reply:
At this point yes. The plugin at the moment is a hack, just so we can render something when opening a PT.

Reply:
Removed.